applications/EigenSolversApplication/custom_solvers/feast_eigensystem_solver.h

```diff
@@ -78,20 +77,102 @@ namespace { // helpers namespace
 
-        KRATOS_INFO_IF( "FEASTEigensystemSolver",
-            mParam["e_min"].GetDouble() != 0.0 || mParam["e_max"].GetDouble() != 0.0 ) <<
-            "Manually defined e_min, e_max are not used for complex symmetric matrices" << std::endl;
-
-        KRATOS_INFO_IF( "FEASTEigensystemSolver",
-            mParam["search_lowest_eigenvalues"].GetBool() || mParam["search_highest_eigenvalues"].GetBool() ) <<
-            "Search for extremal eigenvalues is only available for Hermitian problems" << std::endl;
+        KRATOS_ERROR_IF( mParam["search_lowest_eigenvalues"].GetBool() || mParam["search_highest_eigenvalues"].GetBool() ) <<
+            "Search for extremal eigenvalues is only available for real symmetric problems" << std::endl;
     }
 
     template<> double SettingsHelper<double>::GetE1() {return mParam["e_min"].GetDouble();}
     template<> std::complex<double> SettingsHelper<std::complex<double>>::GetE1() {return std::complex<double>(mParam["e_mid_re"].GetDouble(), mParam["e_mid_im"].GetDouble());}
 
     template<>double SettingsHelper<double>::GetE2() {return mParam["e_max"].GetDouble();}
     template<> double SettingsHelper<std::complex<double>>::GetE2() {return mParam["e_r"].GetDouble();}
+
+    template<typename TScalar>
+    struct SortingHelper
+    {
+        SortingHelper(std::string Order) : mOrder(Order) {};
+        // void Check();
+        template<typename MatrixType, typename VectorType>
+        void SortEigenvalues(VectorType&, MatrixType&);
+        private:
+            std::string mOrder;
+    };
+
+    template<> template<typename MatrixType, typename VectorType>
+    void SortingHelper<double>::SortEigenvalues(VectorType &rEigenvalues, MatrixType &rEigenvectors)
+    {
+        KRATOS_WARNING_IF("FeastEigensystemSolver", mOrder == "si") << "Attempting to sort by imaginary value. Falling back on \"sr\"" << std::endl;
+        KRATOS_WARNING_IF("FeastEigensystemSolver", mOrder == "li") << "Attempting to sort by imaginary value. Falling back on \"lr\"" << std::endl;
+
+        std::vector<size_t> idx(rEigenvalues.size());
+        std::iota(idx.begin(), idx.end(), 0);
+
+        if( mOrder == "sr" || mOrder == "si" ) {
+            std::stable_sort(idx.begin(), idx.end(),
+                [&rEigenvalues](size_t i1, size_t i2) {return rEigenvalues[i1] < rEigenvalues[i2];});
+        } else if( mOrder == "sm") {
+            std::stable_sort(idx.begin(), idx.end(),
+                [&rEigenvalues](size_t i1, size_t i2) {return std::abs(rEigenvalues[i1]) < std::abs(rEigenvalues[i2]);});
+        } else if( mOrder == "lr" || mOrder == "li" ) {
+            std::stable_sort(idx.begin(), idx.end(),
+                [&rEigenvalues](size_t i1, size_t i2) {return rEigenvalues[i1] > rEigenvalues[i2];});
+        } else if( mOrder == "lm") {
+            std::stable_sort(idx.begin(), idx.end(),
+                [&rEigenvalues](size_t i1, size_t i2) {return std::abs(rEigenvalues[i1]) > std::abs(rEigenvalues[i2]);});
```

Comment:
```suggestion
                [&rEigenvalues](std::size_t i1, std::size_t i2) {return std::abs(rEigenvalues[i1]) > std::abs(rEigenvalues[i2]);});
```

Reply:
I don't know why I keep forgetting this..